waiter/src/waiter/process_request.clj

```diff
@@ -284,17 +266,21 @@
 
 (defn inspect-for-202-async-request-response
   "Helper function that inspects the response and triggers async-request post processing."
-  [post-process-async-request-response-fn instance-request-properties service-id metric-group instance endpoint request
-   reason-map {:keys [headers status]} response-headers reservation-status-promise]
+  [{:keys [headers status] :as response} post-process-async-request-response-fn instance-request-properties
+   service-id metric-group instance endpoint request reason-map reservation-status-promise]
   (let [location-header (str (get headers "location"))
         location (async-req/normalize-location-header endpoint location-header)]
-    (when (= status 202)
+    (if (= status 202)
       (if (str/starts-with? location "/")
         (do
           (deliver reservation-status-promise :success-async) ;; backend is processing as an asynchronous request
-          (post-process-async-request-response-fn service-id metric-group instance (handler/make-auth-user-map request)
-                                                  reason-map instance-request-properties location response-headers))
-        (log/info "response status 202, not treating as an async request as location is" location)))))
+          (post-process-async-request-response-fn response service-id metric-group instance
+                                                  (handler/make-auth-user-map request) reason-map
```

Comment:
`(handler/make-auth-user-map request)` has no status/body/headers. `post-process-async-request-response` calls this `response`. Should we interpret any map as a response? Should we call this something else?

Reply:
We're calling `post-process-async-request-response-fn`, defined in core, and it throws this argument away.  Should we clean that up in this PR?

Reply:
`post-process-async-request-response` seems to be using `response` https://github.com/twosigma/waiter/blob/888bbdf70f1c30755469f5dba5631b3e6e89e37e/waiter/src/waiter/async_request.clj#L176

Reply:
I'm still not sure I follow.  `post-process-async-request-response` processes the `response` from the backend.  It needs to look at the `response` to determine if it needs to modify the location.  As an example, `stream-http-response` also has the `response` as the first argument.  Are you saying we still should have the `response` argument at the end?

Reply:
The comment was whether to call this `response`.